ReloadSecureSettingsWithPasswordProtectedKeystoreRestIT failures in fips mode

<!--
Please fill out the following information, and ensure you have attempted
to reproduce locally
-->

**Build scan**:

https://gradle-enterprise.elastic.co/s/27nxkh3jeyqzm/console-log?task=:x-pack:qa:password-protected-keystore:integTest

**Repro line**:

```
./gradlew ':x-pack:qa:password-protected-keystore:integTest' --tests "org.elasticsearch.password_protected_keystore.ReloadSecureSettingsWithPasswordProtectedKeystoreRestIT.testReloadSecureSettingsWithEmptyPassword" -Dtests.seed=779930CF4BC9CD9C -Dtests.security.manager=true -Dtests.locale=sl-SI -Dtests.timezone=America/Hermosillo -Druntime.java=11 -Dtests.fips.enabled=true |  
```



**Reproduces locally?**: yes

**Applicable branches**:

**Failure history**:
<!--
Link to build stats and possible indication of when this started failing and how often it fails
<https://build-stats.elastic.co/app/kibana>
-->
**Failure excerpt**:


```
org.elasticsearch.password_protected_keystore.ReloadSecureSettingsWithPasswordProtectedKeystoreRestIT > testReloadSecureSettingsWithEmptyPassword FAILED
    java.lang.AssertionError: 
    Expected: ("Provided keystore password was incorrect" or "Keystore has been corrupted or tampered with")
         but: was "Error generating an encryption key from the provided password"
        at __randomizedtesting.SeedInfo.seed([779930CF4BC9CD9C:39F40225962A5FBB]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.password_protected_keystore.ReloadSecureSettingsWithPasswordProtectedKeystoreRestIT.testReloadSecureSettingsWithEmptyPassword(ReloadSecureSettingsWithPasswordProtectedKeystoreRestIT.java:80)
```


An exception with "Error generating an encryption key from the provided password" message can now  be thrown in fips mode when a password is short so we need to adjust the matcher when in fips mode

--- a/x-pack/qa/password-protected-keystore/src/test/java/org/elasticsearch/password_protected_keystore/ReloadSecureSettingsWithPasswordProtectedKeystoreRestIT.java
+++ b/x-pack/qa/password-protected-keystore/src/test/java/org/elasticsearch/password_protected_keystore/ReloadSecureSettingsWithPasswordProtectedKeystoreRestIT.java
@@ -15,6 +15,7 @@ import org.elasticsearch.test.rest.ESRestTestCase;
 
 import static org.elasticsearch.xpack.core.security.authc.support.UsernamePasswordToken.basicAuthHeaderValue;
 import static org.hamcrest.Matchers.anyOf;
+import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.instanceOf;
 import static org.hamcrest.Matchers.nullValue;
@@ -65,7 +66,6 @@ public class ReloadSecureSettingsWithPasswordProtectedKeystoreRestIT extends ESR
     }
 
     @SuppressWarnings("unchecked")
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/66951")
     public void testReloadSecureSettingsWithEmptyPassword() throws Exception {
         final Request request = new Request("POST", "_nodes/reload_secure_settings");
         final Response response = client().performRequest(request);
@@ -78,10 +78,16 @@ public class ReloadSecureSettingsWithPasswordProtectedKeystoreRestIT extends ESR
             assertThat(entry.getValue(), instanceOf(Map.class));
             final Map<String, Object> node = (Map<String, Object>) entry.getValue();
             assertThat(node.get("reload_exception"), instanceOf(Map.class));
-            assertThat(ObjectPath.eval("reload_exception.reason", node), anyOf(
-                equalTo("Provided keystore password was incorrect"),
-                equalTo("Keystore has been corrupted or tampered with")));
-            assertThat(ObjectPath.eval("reload_exception.type", node), equalTo("security_exception"));
+            if (inFipsJvm()) {
+                assertThat(ObjectPath.eval("reload_exception.reason", node), containsString(
+                    "Error generating an encryption key from the provided password"));
+                assertThat(ObjectPath.eval("reload_exception.type", node), equalTo("general_security_exception"));
+            } else {
+                assertThat(ObjectPath.eval("reload_exception.reason", node), anyOf(
+                    equalTo("Provided keystore password was incorrect"),
+                    equalTo("Keystore has been corrupted or tampered with")));
+                assertThat(ObjectPath.eval("reload_exception.type", node), equalTo("security_exception"));
+            }
         }
     }
 

--- a/x-pack/qa/password-protected-keystore/src/test/java/org/elasticsearch/password_protected_keystore/ReloadSecureSettingsWithPasswordProtectedKeystoreRestIT.java
+++ b/x-pack/qa/password-protected-keystore/src/test/java/org/elasticsearch/password_protected_keystore/ReloadSecureSettingsWithPasswordProtectedKeystoreRestIT.java
@@ -78,16 +78,15 @@ public class ReloadSecureSettingsWithPasswordProtectedKeystoreRestIT extends ESR
             assertThat(entry.getValue(), instanceOf(Map.class));
             final Map<String, Object> node = (Map<String, Object>) entry.getValue();
             assertThat(node.get("reload_exception"), instanceOf(Map.class));
-            if (inFipsJvm()) {
-                assertThat(ObjectPath.eval("reload_exception.reason", node), containsString(
-                    "Error generating an encryption key from the provided password"));
-                assertThat(ObjectPath.eval("reload_exception.type", node), equalTo("general_security_exception"));
-            } else {
-                assertThat(ObjectPath.eval("reload_exception.reason", node), anyOf(
-                    equalTo("Provided keystore password was incorrect"),
-                    equalTo("Keystore has been corrupted or tampered with")));
-                assertThat(ObjectPath.eval("reload_exception.type", node), equalTo("security_exception"));
-            }
+            assertThat(ObjectPath.eval("reload_exception.reason", node), anyOf(
+                equalTo("Provided keystore password was incorrect"),
+                equalTo("Keystore has been corrupted or tampered with"),
+                containsString("Error generating an encryption key from the provided password") // FIPS
+            ));
+            assertThat(ObjectPath.eval("reload_exception.type", node), 
+                // Depends on exact security provider (eg Sun vs BCFIPS)
+                anyOf(equalTo("security_exception"), equalTo("general_security_exception"))
+            );
         }
     }
 

